pkg/limitador/k8s_objects_test.go

```diff
@@ -58,41 +58,53 @@ func newTestLimitadorObj(name, namespace string, limits []limitadorv1alpha1.Rate
 			},
 		},
 	}
+	if addLabel {
+		limitador.Labels = map[string]string{
+			"test-label-key": "test-label-value",
+		}
+	}
+	return limitador
 }
 
 func TestServiceName(t *testing.T) {
-	name := ServiceName(newTestLimitadorObj("my-limitador-instance", "default", nil))
+	name := ServiceName(newTestLimitadorObj("my-limitador-instance", "default", false, nil))
 	assert.Equal(t, name, "limitador-my-limitador-instance")
 }
 
 func TestDeployment(t *testing.T) {
 	t.Run("default replicas", func(subT *testing.T) {
-		limObj := newTestLimitadorObj("some-name", "some-ns", nil)
+		limObj := newTestLimitadorObj("some-name", "some-ns", false, nil)
 		limObj.Spec.Replicas = nil
 		deployment := Deployment(limObj, DeploymentOptions{})
 		assert.Assert(subT, deployment.Spec.Replicas != nil)
 		assert.Assert(subT, *deployment.Spec.Replicas == 1)
 	})
 
 	t.Run("replicas", func(subT *testing.T) {
-		limObj := newTestLimitadorObj("some-name", "some-ns", nil)
+		limObj := newTestLimitadorObj("some-name", "some-ns", false, nil)
 		limObj.Spec.Replicas = ptr.To(2)
 		deployment := Deployment(limObj, DeploymentOptions{})
```

Comment:
not familiar with limitador but should we assert no labels added in this case?

Reply:
i'm also not that familiar with limitador. 😂   but it makes sense to add that as a test case. 👍